test/tests/functional/pbs_pbsnodes_output_trimmed.py

```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2016 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from ptl.utils.pbs_testsuite import *
+import os
+import subprocess
+from subprocess import Popen, PIPE
+
+
+class Test_pbsnodes_output_trimmed(PBSTestSuite):
+    """
+    This test suite tests pbsnodes executable with -l
+    and makes sure that the node names are not trimmed.
+    """
+
+    def test_pbsnodes_output(self):
+        """
+        This method adds a new entry to /etc/hosts and creates
+        a mom with name more than 20 characters, then makes the
+        node offline and checks the pbsnodes -l output and
+        makes sure it is not trimmed
+        """
+        pbsnodes = os.path.join(
+            str(self.server.pbs_conf['PBS_EXEC']), "bin", "pbsnodes")
+        hname = [
+            '123456789012345678901234567890',
+            '12345678901234567890',
+            '123456789012345']
+        tmpip = ['127.0.0.1', '127.0.0.2', '127.0.0.3']
+        hosts = open("/etc/hosts", "a")
```

Comment:
@rampranesh I don't think we should be modifying /etc/hosts on a machine. If someone cancels the test while it is running, it will just leave this modified hosts file on the system.
I think you can do one of the following:
- Instead of automating it, just add manual steps in the ticket and that to me is fine given that you need a hostname bigger than 20 chars
- If you decide to go with ptl then expect the mom's hostname to be more than 20 chars, if not then just skip the test.
- I was also thinking to use HOSTALIASES env variable but I don't think that it will work because the server would already be running by the time you will set this env variable